extensions/binding/org.eclipse.smarthome.binding.astro/src/main/java/org/eclipse/smarthome/binding/astro/discovery/AstroDiscoveryService.java

```diff
@@ -119,4 +132,22 @@ protected void unsetLocationProvider(LocationProvider locationProvider) {
         this.locationProvider = null;
     }
 
+    @Reference
+    protected void setLocaleProvider(final LocaleProvider localeProvider) {
```

Comment:
These methods seem to be common for all i18n discovery services. Couldn't you move them to a new class "AbstractI18nDiscoveryService" or something like that from which the actual implementation inherits? This would reduce code duplication.

Reply:
It will require to declare a new specific service that will reference itself the required services. Then each binding discovery service that needs i18n will use this service.
@kai suggested to not create a new specific service for this feature, so I tried to use/enhance the existing binding discovery services.

Reply:
Unfortunately, it seems not possible to inherit annotations from a super class.
If it was possible, I could have moved this to the new abstract class.